### ISSUE TITLE
Import PGP secret key?

### ISSUE THREAD

**Issue by feklee** [romanz/trezor-agent#253]
*It’s not possible to import an existing secret key, or is it?*

If so, perhaps that could be documented. Also, if not possible, I wonder if there is a change that key import gets implemented in the future.

**Comment by romanz** [comment]
> It’s not possible to import an existing secret key, or is it?

Yes, it is currently impossible - since the secret keys are cryptographically derived (using BIP-32) from the master secret on your device.

For GPG, you can add your TREZOR-based key as a subkey to your already-existing GPG key, using the `--subkey` flag:
https://github.com/romanz/trezor-agent/blob/master/doc/README-GPG.md#generate-gnupg-subkeys

**Comment by feklee** [comment]
Thanks for clarification!

*Where’s the limitation? trezor-agent, firmware, or hardware?*

I was hoping to upload my encryption key to the Trezor. Guess, I’ll have to stick with smart cards for a while.

**Comment by romanz** [comment]
You're welcome :)

The limitation is the TREZOR firmware design principle (all secret keys must be deterministically derived from a single mnemonic, for backup/recovery purposes), which I actually quite agree with.

But I think that you can extend your current GPG key with new encryption/signature keys, which will be derived from your TREZOR.
Please let me know if the [following link](https://github.com/romanz/trezor-agent/blob/master/doc/README-GPG.md#generate-gnupg-subkeys) helps you.

**Comment by feklee** [comment]
Thanks again!

I primarily use PGP for encryption of files with sensitive information. Switching from the current key would require re-encrypting dozens of files. This could of course be done with the help of a shell script. Creating sub keys, as explained in the [link][1] that you gave me, is an option. I could also just make the key on the Trezor my main key, don’t like messing with multiple keys.

My criteria are:

  * Possibility to restore the key when the Trezor gets broken or lost: *check* (recovery seed + time stamp)

  * Security at least comparable to that of my current 4096 bit RSA key: *to be found out*

[1]: https://github.com/romanz/trezor-agent/blob/master/doc/README-GPG.md#generate-gnupg-subkeys

**Comment by jhoenicke** [comment]
You can add a new signing and/or encryption subkeys to an existing key.  There are some limitations, though.  For example, the sender chooses the newest non-revoked encryption subkey. So if your partner did not update your public key, he may encrypt it with the old key.

There is no easy way to get the GPG key off the Trezor. You'd need to use some special software and your 24 backup words to compute the private key. This would be necessary if you lose your Trezor, or if you want to use gpg without the Trezor.

Trezor doesn't want to store custom GPG keys.  In principle it has enough memory, but one has to think about backup and also it would require a change to the storage sector (and there is the question, how many private GPG keys should it store and how to address them).  Also Trezor can only use elliptic curve cryptography, so DSA and RSA keys wouldn't work anyway.

**Comment by feklee** [comment]
Thanks about the update on security! I also read that - compared to RSA - ECC is easier to crack by Shor's algorithm, i.e. quantum computers. (not that this is an issue any time soon...)
